src/policy/policy.cpp

```diff
@@ -169,16 +169,28 @@ bool AreInputsStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
         std::vector<std::vector<unsigned char> > vSolutions;
         txnouttype whichType = Solver(prev.scriptPubKey, vSolutions);
         if (whichType == TX_NONSTANDARD) {
+            reason = "bad-txns-input-script-nonstandard";
+            debug = strprintf("input %u", i);
             return false;
         } else if (whichType == TX_SCRIPTHASH) {
             std::vector<std::vector<unsigned char> > stack;
             // convert the scriptSig into a stack, so we can inspect the redeemScript
-            if (!EvalScript(stack, tx.vin[i].scriptSig, SCRIPT_VERIFY_NONE, BaseSignatureChecker(), SigVersion::BASE))
+            ScriptError serror;
+            if (!EvalScript(stack, tx.vin[i].scriptSig, SCRIPT_VERIFY_NONE, BaseSignatureChecker(), SigVersion::BASE, &serror)) {
+                reason = "bad-txns-input-scriptsig-failure";
```

Comment:
The order of words typically indicates subcategories/relationships. For that reason, "bad-txns-scriptsig-p2sh-malformed" would fit better.

Reply:
I’m currently using: “bad-txns-input-p2sh-scriptsig-malformed” for this.

Because all reasons in this function relate to inputs, of which are tested/subdivided according to type (p2sh in this case), then the subject scrptsig, then the issue.

Not that there’s a right or wrong here, but that’s the rationale.